lib/src/main/java/graphql/nadel/validation/NadelHydrationValidation.kt

```diff
@@ -274,16 +282,34 @@ internal class NadelHydrationValidation(
                         Locale.getDefault()
                     )
                 ) {
-                    return listOf(NadelSchemaValidationError.StaticArgIsNotAssignable(
-                        parent,
-                        overallField,
-                        remoteArgDef,
-                        actorFieldArg.type,
-                        actorField.name
-                    ))
+                    return listOf(
+                        NadelSchemaValidationError.StaticArgIsNotAssignable(
+                            parent,
+                            overallField,
+                            remoteArgDef,
+                            actorFieldArg.type,
+                            actorField.name
+                        )
+                    )
                 }
                 return emptyList()
             }
         }
     }
+
+    private fun validateConditionsOnAllHydrations(
+        hydrations: List<UnderlyingServiceHydration>,
+        parent: NadelServiceSchemaElement,
+        overallField: GraphQLFieldDefinition,
+    ): NadelSchemaValidationError.SomeHydrationsHaveMissingConditions? {
+        //if one hydration has a condition, then they all must, so lets check the first one
+        var shouldHaveConditions = true
+        if (hydrations.first().conditionalHydration == null) {
+            shouldHaveConditions = false
+        }
+        if (hydrations.any { (it.conditionalHydration == null) == shouldHaveConditions }) { //i.e. (if it has no condition but it should have condition) OR  (if it has condition but shouldn't have condition)
```

Comment:
Hmm, I am not so sure about this logic.

I think this test case would break it

```
@hydrated()
@hydrated(when: )
@hydrated()
```

You can't just check any of them, you need to check all of them.

i.e. logic should be if any `when` exists, then all must have `when`

Reply:
ahh yeah, good catch. I was tryna keep it a small amount of lines but it backfired a little.
changed the logic now to handle that